src/modules/gradle.rs

```diff
@@ -0,0 +1,243 @@
+                            &context
+                                .exec_cmds_return_first(vec![
+                                    vec!["./gradlew", "--version"],
+                                    vec!["gradle", "--version"],
+                                ])?
+                                .stdout,
+                        )?,
+                        GradleVersionStrategy::WrapperProperties => {
+                            let properties_file =
+                                find_wrapper_properties_file(&context.current_dir)?;
+                            let properties = fs::read_to_string(properties_file).ok()?;
+                            parse_gradle_version_from_properties(&properties)?
+                        }
+                    };
+                    VersionFormatter::format_module_version(
+                        module.get_name(),
+                        &gradle_version,
+                        config.version_format,
+                    )
+                    .map(Ok)
+                }
+                _ => None,
+            })
+            .parse(None, Some(context))
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `gradle`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn parse_gradle_version_from_stdout(gradle_stdout: &str) -> Option<String> {
+    // example gradle --version output
+    /*
+       ------------------------------------------------------------
+       Gradle 7.5.1
+       ------------------------------------------------------------
+
+       Build time:   2022-08-05 21:17:56 UTC
+       Revision:     d1daa0cbf1a0103000b71484e1dbfe096e095918
+
+       Kotlin:       1.6.21
+       Groovy:       3.0.10
+       Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
+       JVM:          17.0.4.1 (Microsoft 17.0.4.1+1-LTS)
+       OS:           Linux 5.4.0-1090-azure amd64
+    */
+    let version = gradle_stdout
+        .split_once("Gradle ")?
+        .1
+        .split_whitespace()
+        .next()?;
+    Some(version.to_string())
+}
+
+fn parse_gradle_version_from_properties(wrapper_properties: &str) -> Option<String> {
+    // example gradle.properties content
+    /*
+        distributionBase=GRADLE_USER_HOME
+        distributionPath=wrapper/dists
+        distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-bin.zip
+        zipStoreBase=GRADLE_USER_HOME
+        zipStorePath=wrapper/dists
+    */
+    let version = wrapper_properties
+        .split_once("distributionUrl=")?
+        .1
+        .split_once("gradle-")?
+        .1
+        .split_once('-')?
+        .0;
+    Some(version.to_string())
+}
+
+/// Tries to find the gradle-wrapper.properties file recursively.
+fn find_wrapper_properties_file(directory: &Path) -> Option<PathBuf> {
```

Comment:
Please avoid searching files recursively, unless the module is configured to do this. For the base level, please try checking the scan results if a `gradle` folder exists.